llvm/lib/CASObjectFormats/SimpleV1.cpp

```diff
@@ -0,0 +1,1217 @@
+                    ArrayRef<const jitlink::Edge *> Edges) {
+  auto B = Builder::startNode(CUB.Schema, KindString);
+  if (!B)
+    return B.takeError();
+
+  encoding::writeVBR8(Edges.size(), B->Data);
+  for (const auto *E : Edges) {
+    // Nest the Edge in block.
+    auto SymbolIndex = CUB.getSymbolIndex(E->getTarget());
+    if (!SymbolIndex)
+      return SymbolIndex.takeError();
+    encoding::writeVBR8(*SymbolIndex, B->Data);
+
+    // FIXME: Some of the fields are not stable.
+    unsigned char Bits = 0;
+    Bits |= E->isKeepAlive();
+    Bits |= E->isRelocation() << 1;
+    encoding::writeVBR8(Bits, B->Data);
+    encoding::writeVBR8(E->getKind(), B->Data);
+    encoding::writeVBR8(E->getOffset(), B->Data);
+    encoding::writeVBR8(E->getAddend(), B->Data);
+    if (E->isRelocation())
+      encoding::writeVBR8(E->getRelocation(), B->Data);
+  }
+
+  return get(B->build());
+}
+
+Error EdgeListRef::materialize(LinkGraphBuilder &LGB,
+                               jitlink::Block &Parent) const {
+  auto Remaining = getData();
+
+  unsigned EdgeSize;
+  auto E = encoding::consumeVBR8(Remaining, EdgeSize);
+  if (E)
+    return E;
+
+  for (unsigned I = 0; I < EdgeSize; ++I) {
+    unsigned SymbolIdx;
+    E = encoding::consumeVBR8(Remaining, SymbolIdx);
+    if (E)
+      return E;
+
+    auto Symbol = LGB.getSymbol(SymbolIdx);
+    if (!Symbol)
+      return Symbol.takeError();
+
+    unsigned char Bits, Kind;
+    unsigned Offset, Addend, Relocation = 0;
+    E = encoding::consumeVBR8(Remaining, Bits);
+    if (!E)
+      E = encoding::consumeVBR8(Remaining, Kind);
+    if (!E)
+      E = encoding::consumeVBR8(Remaining, Offset);
+    if (!E)
+      E = encoding::consumeVBR8(Remaining, Addend);
+    if (E)
+      return E;
+
+    bool IsRelocation = Bits & (1U << 1);
+    if (IsRelocation)
+      E = encoding::consumeVBR8(Remaining, Relocation);
+    if (E)
+      return E;
+
+    Parent.addEdge((jitlink::Edge::Kind)Kind, Offset, **Symbol, Addend);
+  }
+  return Error::success();
+}
+
+cl::opt<bool> NestEdgesInBlock("nest-edges-in-block",
+                               cl::desc("Put edges in block"), cl::init(false));
+
+Expected<BlockRef> BlockRef::create(CompileUnitBuilder &CUB,
+                                    const jitlink::Block &Block) {
+  Expected<Builder> B = Builder::startNode(CUB.Schema, KindString);
+  if (!B)
+    return B.takeError();
+
+  // Encode Section.
+  auto SectionIndex = CUB.getSectionIndex(Block.getSection());
+  if (!SectionIndex)
+    return SectionIndex.takeError();
+  encoding::writeVBR8(*SectionIndex, B->Data);
+
+  // Encode block attributes.
+  encoding::writeVBR8(Block.getSize(), B->Data);
+  encoding::writeVBR8(Block.getAlignment(), B->Data);
+  encoding::writeVBR8(Block.getAlignmentOffset(), B->Data);
+  unsigned char Bits = 0;
+  Bits |= Block.isZeroFill();
+  // Not currently useful.
+  Bits |= Block.isContentMutable() << 1;
+  Bits |= Block.isAbsolute() << 2;
```

Comment:
This is always true for `jitlink::Block`, unless I'm reading the code wrong.

`Addressable::isAbsolute()` only returns `false` for external and absolute symbols. But they are not defined and do not have blocks.